Bottom: a74231e1704565ccb0523add022623d5d7f75181
Top:    77c12b69480e7ab9662c7accfa64600dde7f38ce
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-16 09:17:44 -0500

Add codebase generator function

The codebase generator takes information about a change, like repository,
branch, and revision, and creates a predictable string describing the code.
This is used to decide upon the builder for a change and also to tie a fork
repository back to its parent for merge requests.


---
diff --git a/roles/buildbot-master/files/master.cfg b/roles/buildbot-master/files/master.cfg
index aa317118..a4a6fefc 100644
--- a/roles/buildbot-master/files/master.cfg
+++ b/roles/buildbot-master/files/master.cfg
@@ -6,6 +6,7 @@
 
 import os
 from buildbot.plugins import secrets, steps, util
+from buildbot.util import giturlparse
 
 os.environ['LONG_RUN_TIMEOUT'] = '14400'
 
@@ -140,3 +141,30 @@ c['workers'].extend(ptxdist.workers)
 c['schedulers'].extend(ptxdist.schedulers)
 c['builders'].extend(ptxdist.builders)
 c['change_source'].extend(ptxdist.change_source)
+
+
+def codebaseGenerator(chdict):
+    """ Generate a codebase description.
+
+    The codebase generator takes information about a change, like repository,
+    branch, and revision, and creates a predictable string describing the code.
+    This is used to decide upon the builder for a change and also to tie a fork
+    repository back to its parent for merge requests.
+    """
+
+    giturl = giturlparse(chdict['repository'])
+    return "%s/%s" % (giturl.owner, giturl.repo)
+
+    # path = urlparse(chdict['revlink']).path.lstrip('/')
+
+    # components = []
+    # while path != "":
+    #     path, tail = os.path.split(path)
+    #     if tail != "":
+    #         components.append(tail)
+    # components.reverse()
+    # codebase = os.path.join(*components[0:2])
+    # return codebase
+
+
+c['codebaseGenerator'] = codebaseGenerator
